api/routes/saved_jobs.py

```
@@ -0,0 +1,316 @@
+
+    try:
+        # Create saved job record - use service client to bypass RLS
+        saved_job_data = {"user_id": user_id, "job_id": job_id, "notes": request.notes}
+
+        response = service_client.table("saved_jobs").insert(saved_job_data).execute()
+
+        if response.data:
+            logger.info(f"User {user_id} saved job {job_id}")
+            return response.data[0]
+        else:
+            raise Exception("Failed to save job")
```

Ruff (0.13.3)

Abstract `raise` to an inner function

(TRY301)